src/components/Grid/Grid.js

```diff
@@ -18,20 +18,21 @@ const GridComponent = ( {
         <>
             <div className="grid-container">
                 {
-                    parsedData.map(({ id, url, name, category, price }) => { return ( 
+                    parsedData.map(({ id, url, name, category, price,stock }) => { return ( 
                         <GridCard 
                             key={id}
                             cardId={id} 
                             url={url}
                             name={name}
                             category={category}
                             price={price}
+                            stock={stock}
                         />) 
                     }) 
                 }
             </div>
             {
-                pagination &&
+                pagination && totalPages > 1 &&
```

Comment:
maybe handle this evaluation on a variable. here you can have just a variable that has the value of `pagination && totalPages > 1 `
```suggestion
                showPagination &&
```